src/message.rs

```diff
@@ -65,7 +91,7 @@ impl TransferBuilder {
     }
 
     /// (Optional) message indexation.
-    pub fn with_indexation(mut self, indexation: Indexation) -> Self {
+    pub fn with_indexation(mut self, indexation: IndexationBuilder) -> Self {
```

Comment:
Should the method name be changed to `with_indexation_builder`?

Reply:
yeah this is looking weird both ways, so I'm just going to rewrite it to have the TransferBuilder type with the original Indexation struct, and deserialize with a wrapper struct.